lib/pipelineTemplateVersionFactory.js

```diff
@@ -162,9 +162,12 @@ class PipelineTemplateVersionFactory extends BaseFactory {
             return [];
         }
 
-        config.params.templateId = pipelineTemplateMeta.id;
-
-        return super.list(config);
+        return super.list({
+            ...config,
+            params: {
+                templateId: pipelineTemplateMeta.id
+            }
```

Comment:
We should not override the params with just `templateId`. There could be other params (Ex: `createTime`) besides `name` and `namesapce`.
Instead, remove `name` and `namespace` from the params

Reply:
not really, `pipelineTemplateVersion` schema does not have name,namespace

Reply:
I agree that we don't need to exclude `name` and `namespace`. But it is still unclear on why we are picking only `templateId` from params.
Model interface should not be designed/restricted due to current use cases in the Screwdriver API. It should allow filtering on any of the `pipelineTemplateVersion` field.

Reply:
the params don't get ignored I think, so we need to remove them.
https://github.com/screwdriver-cd/models/blob/master/lib/baseFactory.js#L132

Reply:
Yes, looks like if there are params not specified in the schema, error is thrown
https://github.com/screwdriver-cd/datastore-sequelize/blob/master/index.js#L475

Then, as originally suggested, let us exclude `name` and `namespace` from `config.params`.

Or, if we want to be strict and reserve `config.params` to only allow the fields of `pipelineTemplateVersion`, then should we have `name` and `namespace` on `config` and not in `config.params`?

Reply:
Yes, I think we can move `name` and `namespace` from `config.params`